### ISSUE TITLE
accessibilityLabel does not work for Text component on iOS

### ISSUE THREAD

**Issue by curioustechizen** [facebook/react-native#14123]
### Description

Continuing the discussion from https://github.com/facebook/react-native/issues/13648#issuecomment-303046218, for `<Text>` component on iOS, the actual label of the text always overrides the `testID` and `accessibilityLabel`

### Reproduction Steps and Sample Code

1. Create a new RN app using `react-native init`
2. Edit index.ios.js
3. Find the "Welcome to React Native" text and change it to 
```
<Text testID={'welcome'} accessibilityLabel={'welcome'}>Welcome to React Native!</Text>
```
4. Run this app on an iOS simulator and inspect it with Accessibility Inspector

**Expected:** The Accessibility Inspector should say the Label is `welcome`.
**Actual:** The Accessibility Inspector says the Label is `Welcome to React Native!`.

_Note:_ - with the same set of properties on Android the `contentDescription` is `welcome` which is what I expect.

### Additional Information

* React Native version: 0.44.0
* Platform: iOS 10.1 Simulator
* Development Operating System: Mac OSX 10.11.6
* Dev tools: XCode 8.1, Accessibility Inspector 5.0

**Comment by shergin** [comment]
Thank you, @curioustechizen! I will fix that.

**Comment by hramos** [comment]
Hi there! This issue is being closed because {:search=>"repo:facebook/react-native is:open is:issue updated:<2017-05-25 -label:\"Good First Task\" -label:\"Help Wanted\" ", :message=>"it has been inactive for a while"}. Maybe the issue has been fixed in a recent release, or perhaps it is not affecting a lot of people. Either way, we're automatically closing issues after a period of inactivity. Please do not take it personally!

If you think this issue should definitely remain open, please let us know. The following information is helpful when it comes to determining if the issue should be re-opened:

- Does the issue still reproduce on the latest release candidate? Post a comment with the version you tested.
- If so, is there any information missing from the bug report? Post a comment with all the information required by the [issue template](https://github.com/facebook/react-native/blob/master/.github/ISSUE_TEMPLATE.md).
- Is there a pull request that addresses this issue? Post a comment with the PR number so we can follow up.

If you would like to work on a patch to fix the issue, *contributions are very welcome*! Read through the [contribution guide](http://facebook.github.io/react-native/docs/contributing.html), and feel free to hop into [#react-native](https://discordapp.com/invite/0ZcbPKXt5bZjGY5n) if you need help planning your contribution.

**Comment by shergin** [comment]
@curioustechizen Could you please verify that it is already fixed? (It should be.)

**Comment by curioustechizen** [comment]
@shergin Does not seem to be fixed. With

* React Native version: **0.46.4**
* Platform: iOS 10.3 Simulator
* Development Operating System: Mac OSX 10.12.5
* Dev tools: XCode 8.3.2, Accessibility Inspector 5.0

I still see 
**Actual:** The Accessibility Inspector says the Label is `Welcome to React Native!`.

Is there a way to programmatically display current RN version on screen - just to make sure I'm not messing something up?

**Comment by Leeds-eBooks** [comment]
@curioustechizen You can use https://github.com/rebeccahughes/react-native-device-info#example to get the current version.

**Comment by curioustechizen** [comment]
@Leeds-eBooks I don't want to display the app version - I want to find out the version of ReactNative that's being used in the app. I couldn't find a way to do that with the linked library.

**Comment by Leeds-eBooks** [comment]
Ah stupid me, sorry! A hacky way would be to `require` the RN `package.json`.

**Comment by vasyares** [comment]
Hello. Any updates about this issue?

**Comment by shergin** [comment]
Sorry, not yet. 😞  (t21623750)

**Comment by vasyares** [comment]
@shergin thank you for the quick response. 
We are using accessibilityLabel for UI testing on our project. Could you please tell us when this issue might be fixed? 
Thank you.

**Comment by shergin** [comment]
We are currently working at it. I will keep posting updates here.

**Comment by vasyares** [comment]
Thank you. 
BTW we started using testId attribute. Seems it works. We are able to see correct identifiers with Debug View Hierarchy after attaching to process with Xcode debugger.

**Comment by vasyares** [comment]
Hello @shergin, 
Looks like issue for text element is still there. Using of testId does not fix it.

**Comment by shergin** [comment]
cc @lostatseajoshua

**Comment by lostatseajoshua** [comment]
@curioustechizen @vasyares The fix has been made and is in a PR. Setting the `accessibilityLabel` of a `<Text>` component will now return the correct value as expected.

**Comment by hramos** [comment]
This has been fixed as of https://github.com/facebook/react-native/commit/915a020fcaaefc8c8c5a2016e79e9d6127c6e008
